src/module.c

```diff
@@ -8420,6 +8438,9 @@ void moduleGILBeforeUnlock() {
      * released we have to propagate here). */
     exitExecutionUnit();
     postExecutionUnitOperations();
+
+    if (listLength(server.ready_keys))
+        handleClientsBlockedOnKeys();
```

Comment:
:+1:

Reply:
Yes I also thought it should be there but was afraid moving it there for the same reason you mentioned.

Reply:
i understand it fells like a risky change, but unblocking clients at the end of an execution unit is the right thing to do

Reply:
i think we should put some effort into investigating the option of adding `handleClientsBlockedOnKeys` to `postExecutionUnitOperations`

Reply:
I agree, I want to look into it to see what that means, have to look at all points where it is called from.